Kickstarter-iOS/Views/Controllers/LiveStreamContainerPageViewController.swift

```diff
@@ -0,0 +1,202 @@
+      .flatMap { $0 as? UIPageViewController }
+      .first
+
+    self.pageViewController?.dataSource = self.pagesDataSource
+    self.pageViewController?.delegate = self
+
+    self.viewModel.inputs.viewDidLoad()
+  }
+
+  internal override func bindStyles() {
+    super.bindStyles()
+
+    _ = self
+      |> baseControllerStyle()
+      |> UIViewController.lens.view.backgroundColor .~ .hex(0x353535)
+
+    _ = self.separatorView
+      |> UIView.lens.backgroundColor .~ UIColor.white.withAlphaComponent(0.2)
+
+    _ = self.indicatorLineView
+      |> UIView.lens.backgroundColor .~ .white
+
+    _ = self.pagerTabStripStackView
+      |> UIStackView.lens.layoutMarginsRelativeArrangement .~ true
+      |> UIStackView.lens.layoutMargins .~ .init(topBottom: Styles.grid(2))
+
+    _ = self.infoPagerButton
+      |> UIButton.lens.title(forState: .normal) .~ localizedString(key: "Info", defaultValue: "Info")
+
+    _ = self.chatPagerButton
+      |> UIButton.lens.title(forState: .normal) .~ localizedString(key: "Chat", defaultValue: "Chat")
+  }
+
+  internal override func bindViewModel() {
+    super.bindViewModel()
+
+    self.viewModel.outputs.loadViewControllersIntoPagesDataSource
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.loadViewControllersIntoPagesDataSource(pages: $0)
+    }
+
+    self.viewModel.outputs.chatButtonTextColor
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.chatPagerButton.setTitleColor($0, for: .normal)
+    }
+
+    self.viewModel.outputs.chatButtonTitleFont
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.chatPagerButton.titleLabel?.font = $0
+    }
+
+    self.viewModel.outputs.infoButtonTextColor
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.infoPagerButton.setTitleColor($0, for: .normal)
+    }
+
+    self.viewModel.outputs.infoButtonTitleFont
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.infoPagerButton.titleLabel?.font = $0
+    }
+
+    self.viewModel.outputs.indicatorLineViewXPosition
+      .observeForUI()
+      .observeValues { [weak self] in
+        self?.animateIndicatorLineViewToPosition(position: $0)
+    }
+
+    self.viewModel.outputs.pagedToPage
+      .observeForUI()
+      .observeValues { [weak self] page, direction in
+        switch page {
+        case .chat:
+          self?.liveStreamChatViewController.flatMap {
+            self?.pageViewController?.setViewControllers([$0], direction: direction, animated: true,
+                                                        completion: nil)
+          }
+        case .info:
+          self?.liveStreamEventDetailsViewController.flatMap {
+            self?.pageViewController?.setViewControllers([$0], direction: direction, animated: true,
+                                                        completion: nil)
+          }
+        }
+    }
+  }
+
+  @objc private func info() {
```

Comment:
Guess I've always just named actions that way from [these](http://nshipster.com/ibaction-iboutlet-iboutletcollection/) suggestions and that we're being more explicit in our view model and tests about exactly what 'event' took place. But all good! updated.